### DISCUSSION TITLE
internet permission in sdk manifest

### DISCUSSION THREAD

**Discussion by Meibes** [ngageoint/geopackage-android#72] [Q&A] [answered]
Hello,

thanks for this awesome library!

I was testing your library in a demo app and noticed that my demo app suddenly requested the "android.permission.INTERNET"  permission, although I did not request it in my manifest. After some research I found out that in the Merged Manifest the permission came from [geopackage-android](https://github.com/ngageoint/geopackage-android/blob/367c1b353917cd6f734af0b3044818d251ccc073/geopackage-sdk/src/main/AndroidManifest.xml#L5). I can understand that you need this permission for [testing purposes](https://github.com/ngageoint/geopackage-android/blob/367c1b353917cd6f734af0b3044818d251ccc073/geopackage-sdk/src/androidTest/AndroidManifest.xml#L5) and in the [mapcache demo app](https://github.com/ngageoint/geopackage-mapcache-android/blob/4961ef99f15d11ab59a2cf3b9afabe6f4be52b7d/mapcache/src/main/AndroidManifest.xml#L6), but is it really needed directly in the sdk? My demo app just works fine, even if I remove the permission for internet, maybe because I'm not using any of the classes that require internet (I don't know which ones actually do).

Could you remove the permission or would that break anything?

greetz

**Comment by bosborn** [comment, marked as answer]
It looks like I can remove the permission from [geopackage-android ](https://github.com/ngageoint/geopackage-android/blob/367c1b353917cd6f734af0b3044818d251ccc073/geopackage-sdk/src/main/AndroidManifest.xml#L5)and [geopackage-android-map](https://github.com/ngageoint/geopackage-android-map/blob/73fcc455ff7d7d2763a5d57205fe8f773d677cb4/geopackage-map/src/main/AndroidManifest.xml#L4) and will do so in the next release.

Any use of network utilities ([import from URL](https://github.com/ngageoint/geopackage-android/blob/367c1b353917cd6f734af0b3044818d251ccc073/geopackage-sdk/src/main/java/mil/nga/geopackage/GeoPackageManagerImpl.java#L860), [URL tile generator](https://github.com/ngageoint/geopackage-android/blob/master/geopackage-sdk/src/main/java/mil/nga/geopackage/tiles/UrlTileGenerator.java)) w/o including the permission will cause a permission denied security exception.

**Reply by bosborn** [reply]
Removed in [geopackage-android 6.2.0](https://github.com/ngageoint/geopackage-android/releases/tag/6.2.0) and [geopackage-android-map 6.2.0](https://github.com/ngageoint/geopackage-android-map/releases/tag/6.2.0).

**Reply by Meibes** [reply]
thank you!
